oeps/oep-0044-bp-changelogs.rst

```diff
@@ -0,0 +1,294 @@
+
+* Incrementing the version appropriately for each PR merge, based on a well formatted commit comment. See the `semantic-release`_ README for details. A pre-commit hook is typically run to ensure the commit message is appropriately formatted for `semantic-release`_.
+
+  * One important formatting detail is to include "BREAKING CHANGE" at the beginning of a line in the commit description to get a new major version.
+
+* Publishing the changelog message as a GitHub Release in the repository, using the commit message. Here is an example `GitHub Release changelog for edx/frontend-build`_.
+
+As noted above, please update the changelogs as necessary to add clarifications or customize the messaging for the audience of the changelog, especially because these messages originally come from commit messages, which may have been written for a different audience.
+
+.. _semantic-release: https://github.com/semantic-release/semantic-release
+.. _GitHub Release changelog for edx/frontend-build: https://github.com/edx/frontend-build/releases
+
+Python Libraries
+~~~~~~~~~~~~~~~~
+
+Like all Open edX repositories, Python libraries should also include a CHANGELOG.rst. This file may also simply point to the GitHub Releases. Here is an example CHANGELOG.rst::
+
+  Changelog
+  =========
+
+  All notable changes to this project will be documented as `GitHub Releases`_.
+
+  This project adheres to `Semantic Versioning`_.
+
+  .. _GitHub Releases: https://github.com/edx/frontend-build/releases
+  .. _Semantic Versioning: https://semver.org/spec/v2.0.0.html
+
+Many Python libraries publish packages to PyPI when a version is tagged. Publishing a GitHub Release both tags the version and allows one to write an appropriate changelog message. This is the recommended process at this time.
+
+Every published release should have a corresponding changelog entry.
+
+Application and Non-Library Repositories
+----------------------------------------
+
+When it comes to changelogs, the main difference between application and non-library repositories, and the eariler recommendations for library repositories, is that non-library code is not versioned.
+
+Since this code has no versions, the recommendations differ in the following ways:
+
+* The changelog entries will live in the CHANGELOG.rst file, following the recommendations detailed in `Changelogs and Changelog Entries`_.
+* Changelog entries should start with a date, rather than a date and version, because these repositories are not versioned.
+
+Changelog for edx-platform
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The edx-platform repository should have a CHANGELOG.rst, just like all other repositories.
+
+* Many notes that could potentially live in the edx-platform CHANGELOG.rst get added to the `Open Edx Release Planning`_ pages which have historically been created for each named release. Worst case, edx-platform notes are manually added to the CHANGELOG.rst instead, and at least the documentation stays more closely tied to the code.
```

Comment:
I don't know what "Worst case" is clarifying?

Reply:
I took two steps here:
1. More importantly, I deleted this whole section which was both speculative and really does belong in an edx-platform ADR when and if any of this would provide help.
2. I updated the language around "worst case".  I am going to copy the text to this comment in case I ever want to see it again, but I think this was mostly about me getting stuff out of my head.

```
Changelog for edx-platform
~~~~~~~~~~~~~~~~~~~~~~~~~~

The edx-platform repository should have a CHANGELOG.rst, just like all other repositories.

* Many notes that could potentially live in the edx-platform CHANGELOG.rst get added to the `Open Edx Release Planning`_ pages which have historically been created for each named release. Instead, adding entries to the edx-platform CHANGELOG.rst would follow the best practice of keeping documentation close to its code.

* Entries could be grouped by day if they could be added at a faster cadence (see `Tooling and Automation`_). Otherwise, date ranges should be used.

* Attempting to add changelog entries at a very faster cadence without automation is likely to cause too many merge conflict issues. However, this should not stop people from adding entries immediately so we can learn.

* As with all changelogs, it is acceptable and appropriate to clean up or enhance or add changelog entries as necessary.

* More work is clearly required to determine how to iterate and improve the process for a CHANGELOG.rst in edx-platform.

* The CHANGELOG.rst will also likely need to be broken into multiple files, but that can be a local decision.

.. _Open Edx Release Planning: https://openedx.atlassian.net/wiki/spaces/COMM/pages/13205845/Open%2BedX%2BRelease%2BPlanning
```